### ISSUE TITLE
Latest version time out issue

### ISSUE THREAD

**Issue by herkalurk** [sabnzbd/sabnzbd#2200]
After latest version (3.6.0) was released, I cannot start sabnzbd service.  The below message just repeats on restarts.  

```
2022-06-07 09:07:10,595::INFO::[SABnzbd:1167] --------------------------------
2022-06-07 09:07:10,595::INFO::[SABnzbd:1168] SABnzbd.py-3.6.0
2022-06-07 09:07:10,595::INFO::[SABnzbd:1178] Commit = b31fedd85790c21488cbebd9730bdc00b2bdb385
2022-06-07 09:07:10,595::INFO::[SABnzbd:1180] Full executable path = /app/sabnzbd/SABnzbd.py
2022-06-07 09:07:10,595::INFO::[SABnzbd:1181] Arguments = "/app/sabnzbd/SABnzbd.py" "--config-file" "/config" "--server" "::"
2022-06-07 09:07:10,596::INFO::[SABnzbd:1182] Python-version = 3.9.7 (default, Nov 24 2021, 21:15:59)
[GCC 10.3.1 20211027]
2022-06-07 09:07:10,596::INFO::[SABnzbd:1183] Dockerized = True
2022-06-07 09:07:10,596::INFO::[SABnzbd:1184] CPU architecture = x86_64
2022-06-07 09:07:10,602::INFO::[SABnzbd:1187] Platform = posix - Linux-3.10.0-1160.66.1.el7.x86_64-x86_64-with
2022-06-07 09:07:10,602::INFO::[SABnzbd:1193] Preferred encoding = UTF-8
2022-06-07 09:07:10,603::DEBUG::[SABnzbd:1212] JSON-module = ujson 5.3.0
2022-06-07 09:07:10,603::INFO::[SABnzbd:1215] SSL version = OpenSSL 1.1.1o  3 May 2022
2022-06-07 09:07:10,604::INFO::[SABnzbd:1224] Certifi version = 2020.12.05
2022-06-07 09:07:10,604::INFO::[SABnzbd:1225] Loaded additional certificates from /etc/ssl/certs/ca-certificates.crt
2022-06-07 09:07:10,623::DEBUG::[SABnzbd:1234] Available certificates = {'x509': 132, 'crl': 0, 'x509_ca': 132}
2022-06-07 09:07:10,623::DEBUG::[getipaddress:100] Local IPv4 address = 172.19.0.2
Traceback (most recent call last):
  File "/app/sabnzbd/SABnzbd.py", line 1786, in <module>
    main()
  File "/app/sabnzbd/SABnzbd.py", line 1238, in main
    publicipv4()
  File "/app/sabnzbd/sabnzbd/getipaddress.py", line 114, in publicipv4
    result = addresslookup4(sabnzbd.cfg.selftest_host())
  File "/app/sabnzbd/sabnzbd/getipaddress.py", line 48, in func_wrapper
    return sabnzbd.THREAD_POOL.submit(item, *args, **kwargs).result(max_timeout)
  File "/usr/lib/python3.9/concurrent/futures/_base.py", line 447, in result
    raise TimeoutError()
concurrent.futures._base.TimeoutError
```

**Comment by sanderjo** [comment]
I think I had that with a 3.6.0 RC ... after a system reboot, everything worked OK. So can you try that first?

**Comment by herkalurk** [comment]
That did not work.  I have sabnzbd running in a docker container, and even after fully restarting the host system it is still showing the same error.

**Comment by sanderjo** [comment]
Interesting. Which docker image are you using?

And on which OS are you running docker?

**Comment by herkalurk** [comment]
Docker image is linuxserver/sabnzbd

The docker host is centos 7, I'm using the docker-ce repo for the docker engine

Up until this version I didn't see this issue.  Have been in a docker container for over a year now with other version changes.

**Comment by sanderjo** [comment]
I **cannot** reproduce on my Ubuntu host OS: `docker run -p 8080:8080 lscr.io/linuxserver/sabnzbd:latest` starts, and the wrench nicely shows the public IPv4

So ... is it caused by your underlying CentOS 7? Long shot ...

Another long shot: what if you start the docker image in privileged mode, so  `docker run --privileged  -p 8080:8080 lscr.io/linuxserver/sabnzbd:latest`

**Comment by herkalurk** [comment]
I just adjusted the docker-compose file and added/removed privileged and that didn't change it.  Another person suggested could it be a legacy setting in the sabnzbd.ini file.  Have there been any changes there?

**Comment by Safihre** [comment]
That's indeed unfortunate bug. 
You can work around this by editing you `sabnzbd.ini` file and setting `logging` to `0`

**Comment by herkalurk** [comment]
Moving my configs out does result in a valid start to the setup wizard.  The logs don't show any timeouts with attempting to determine the public IP.  Which settings affect the public ip?

```
2022-06-07 10:45:22,470::INFO::[SABnzbd:1167] --------------------------------
2022-06-07 10:45:22,470::INFO::[SABnzbd:1168] SABnzbd.py-3.6.0
2022-06-07 10:45:22,470::INFO::[SABnzbd:1178] Commit = b31fedd85790c21488cbebd9730bdc00b2bdb385
2022-06-07 10:45:22,471::INFO::[SABnzbd:1180] Full executable path = /app/sabnzbd/SABnzbd.py
2022-06-07 10:45:22,471::INFO::[SABnzbd:1181] Arguments = "/app/sabnzbd/SABnzbd.py" "--config-file" "/config" "--server" "::"
2022-06-07 10:45:22,471::INFO::[SABnzbd:1182] Python-version = 3.9.7 (default, Nov 24 2021, 21:15:59)
[GCC 10.3.1 20211027]
2022-06-07 10:45:22,471::INFO::[SABnzbd:1183] Dockerized = True
2022-06-07 10:45:22,471::INFO::[SABnzbd:1184] CPU architecture = x86_64
2022-06-07 10:45:22,478::INFO::[SABnzbd:1187] Platform = posix - Linux-3.10.0-1160.66.1.el7.x86_64-x86_64-with
2022-06-07 10:45:22,478::INFO::[SABnzbd:1193] Preferred encoding = UTF-8
2022-06-07 10:45:22,478::INFO::[SABnzbd:1215] SSL version = OpenSSL 1.1.1o  3 May 2022
2022-06-07 10:45:22,480::INFO::[SABnzbd:1224] Certifi version = 2020.12.05
2022-06-07 10:45:22,480::INFO::[SABnzbd:1225] Loaded additional certificates from /etc/ssl/certs/ca-certificates.crt
2022-06-07 10:45:22,480::INFO::[SABnzbd:1246] Using INI file /config/sabnzbd.ini
2022-06-07 10:45:22,483::INFO::[postproc:134] Loading postproc queue
2022-06-07 10:45:22,676::INFO::[scheduler:188] Scheduling RSS interval task every 60 min (delay=59)
2022-06-07 10:45:22,676::INFO::[scheduler:199] Scheduling VersionCheck on day 2 at 19:50
2022-06-07 10:45:22,677::INFO::[scheduler:213] Setting schedule for midnight BPS reset
2022-06-07 10:45:22,677::INFO::[scheduler:216] Setting schedule for server expiration check
2022-06-07 10:45:22,677::INFO::[scheduler:221] Setting schedule for server quota check
2022-06-07 10:45:22,678::INFO::[__init__:341] All processes started
2022-06-07 10:45:22,678::INFO::[SABnzbd:336] Template location for Glitter is /app/sabnzbd/interfaces/Glitter
2022-06-07 10:45:22,678::INFO::[SABnzbd:336] Template location for Config is /app/sabnzbd/interfaces/Config
2022-06-07 10:45:22,680::INFO::[misc:1163] [N/A] Running external command: ['/usr/local/bin/unrar']
2022-06-07 10:45:22,693::INFO::[misc:1163] [N/A] Running external command: ['/usr/bin/7za']
2022-06-07 10:45:22,700::INFO::[misc:1163] [N/A] Running external command: ['/usr/bin/par2', '-h']
2022-06-07 10:45:22,707::INFO::[SABnzbd:423] SABYenc module (v5.4.2)... found!
2022-06-07 10:45:22,707::INFO::[SABnzbd:424] SABYenc module is using SIMD set: SSSE3
2022-06-07 10:45:22,708::INFO::[SABnzbd:443] Cryptography module (v37.0.2)... found!
2022-06-07 10:45:22,708::INFO::[SABnzbd:448] par2 binary... found (/usr/bin/par2)
2022-06-07 10:45:22,708::INFO::[SABnzbd:455] UNRAR binary... found (/usr/local/bin/unrar)
2022-06-07 10:45:22,709::INFO::[SABnzbd:463] UNRAR binary version 6.10
2022-06-07 10:45:22,710::INFO::[SABnzbd:471] 7za binary... found (/usr/bin/7za)
2022-06-07 10:45:22,710::INFO::[SABnzbd:473] 7za binary version 16.02
2022-06-07 10:45:22,710::INFO::[SABnzbd:484] nice binary... found (/bin/nice)
2022-06-07 10:45:22,711::INFO::[SABnzbd:488] ionice binary... found (/bin/ionice)
2022-06-07 10:45:22,713::INFO::[SABnzbd:1435] Starting web-interface on :::8080
2022-06-07 10:45:22,714::INFO::[_cplogging:213] [07/Jun/2022:10:45:22] ENGINE Bus STARTING
/usr/lib/python3.9/site-packages/cherrypy/process/servers.py:416: UserWarning: Unable to verify that the server is bound on 8080
  warnings.warn(msg)
2022-06-07 10:45:27,849::INFO::[_cplogging:213] [07/Jun/2022:10:45:27] ENGINE Serving on http://:::8080
2022-06-07 10:45:27,849::INFO::[_cplogging:213] [07/Jun/2022:10:45:27] ENGINE Bus STARTED
2022-06-07 10:45:27,849::INFO::[SABnzbd:1474] Starting SABnzbd.py-3.6.0
2022-06-07 10:45:27,854::INFO::[dirscanner:117] Dirscanner starting up
2022-06-07 10:45:27,855::INFO::[panic:239] Launching browser with http://127.0.0.1:8080/sabnzbd
2022-06-07 10:45:27,857::INFO::[notifier:123] Sending notification: SABnzbd - SABnzbd 3.6.0 started (type=startup, job_cat=None)
2022-06-07 10:45:32,133::INFO::[zconfig:61] No bonjour/zeroconf support installed
2022-06-07 10:45:32,134::INFO::[ssdp:108] Serving SSDP on 172.19.0.5 as SABnzbd
2022-06-07 10:47:06,243::INFO::[config:902] Writing settings to INI file /config/sabnzbd.ini
```

**Comment by herkalurk** [comment]
@Safihre Below is the logging section in the ini file, which part should I disable?

```
[logging]
log_new = 0
max_log_size = 5242880
log_level = 2
log_backups = 5
enable_cherrypy_logging = 0
```

**Comment by Safihre** [comment]
Change to `log_level = 0`

**Comment by herkalurk** [comment]
> Change to `log_level = 0`

I changed to 0 and was able to start, I was also able to start at `log_level =1`

**Comment by Safihre** [comment]
Seems something is special about your setup, since the webserver also can't check if it's up. So seems the Docker is not allowed to connect outside? You have some special firewall running?

**Comment by sanderjo** [comment]
> Seems something is special about your setup, since the webserver also can't check if it's up. So seems the Docker is not allowed to connect outside? You have some special firewall running?

It might be @herkalurk started that test docker instance without `-p 8080:8080`. In my experience, SAB then gives that warning

**Comment by sanderjo** [comment]
> That's indeed unfortunate bug.
> You can work around this by editing you `sabnzbd.ini` file and setting `logging` to `0`

Can you explain what the bug is? How can log level impact the public ipv4 detection? I cannot reproduce with any log level...

**Comment by sanderjo** [comment]
> Yes, I am running http port on 8085.

Let me refrase then: 
with the workaround and your real sabnzbd.ini, can you access SAB from outside?
with the workaround and your real sabnzbd.ini, is that line still there in your logging?

```
2022-06-07 10:45:22,714::INFO::[_cplogging:213] [07/Jun/2022:10:45:22] ENGINE Bus STARTING
/usr/lib/python3.9/site-packages/cherrypy/process/servers.py:416: UserWarning: Unable to verify that the server is bound on 8080
  warnings.warn(msg)
```

**Comment by Safihre** [comment]
@sanderjo at Debug level we check the external IP at startup by connecting to the Self-test host. But in the case of @herkalurk this results in a timeout. We should catch that and return `None`, but we don't catch it.
By not setting to Debug, the IP check is not performed.
@herkalurk probably has the same crash if he open the Status and Interface settings window.

**Comment by sanderjo** [comment]
I can reproduce the traceback with "only local network/Wifi connection (so an IP address plus default gateway), but no Internet connectivity"
@herkalurk can that be the case for your setup? In other words: do you or don't you have Internet connectivity from your SAB-in-docker to Internet?

**Comment by herkalurk** [comment]
> @herkalurk can that be the case for your setup? In other words: do you or don't you have Internet connectivity from your SAB-in-docker to Internet?

The docker container can get to the internet, otherwise I couldn't actually get to the nntp servers.  After I changed my log level sonarr submitted a job and it successfully downloaded all the articles in the nzb file.  The log is also showing valid connections to the rss feeds I'm reading.

```
2022-06-07 12:21:46,522::INFO::[rss:473] Starting scheduled RSS read-out for "carts"
2022-06-07 12:21:50,784::INFO::[rss:234] RSS Feed https://newz69.keagaming.com/rss?t=-2&dl=1&i=1&r=KEY&del=1 was empty
2022-06-07 12:22:01,339::INFO::[rss:234] RSS Feed https://api.nzb.su/rss?t=-2&dl=1&i=53497&r=KEY&del=1 was empty
2022-06-07 12:22:16,396::INFO::[rss:489] Finished scheduled RSS read-outs
```

**Comment by sanderjo** [comment]
Also in `result = addresslookup4(sabnzbd.cfg.selftest_host())`

That is only the DNS lookup of sabnzbd.cfg.selftest_host(). How can that go wrong with a Timeout? No or slow Internet, no or slow dns lookup, ... ?

Anyway: a patch is in progress https://github.com/sabnzbd/sabnzbd/pull/2202

**Comment by RafaelSchridi** [comment]
DNS lookup inside the container is working fine, internet is on 1gbit fiber. I do have a it behind a PiHole instance, could that be causing problems?

**Comment by Safihre** [comment]
Strange. What does INSECURE mean in this case?

**Comment by sanderjo** [comment]
@RafaelSchridi what is your setup? Like: is your SAB in docker, or on a plain OS?

**Comment by sanderjo** [comment]
@herkalurk are you using PiHole too?

**Comment by sanderjo** [comment]
both of you:

what if you visit https://self-test.sabnzbd.org/ with your web browser?
what is the result of
`host self-test.sabnzbd.org`
or
`nslookup self-test.sabnzbd.org`

**Comment by sanderjo** [comment]
If you possible, can you both do this? If not on Docker, it's less work.

**Docker**

Start SABnzbd in docker with name 'sabnzbd'
```
sander@witte2004:~$ docker run -d --name sabnzbd -p 8080:8080 lscr.io/linuxserver/sabnzbd:latest 
4d549f2f6a9c3f5671e4ccdc935372ae89f8ab1440d7ec5c3994d31f20d7d944

```

Go into the running docker image, and execute commands

```
sander@witte2004:~$ docker exec -it sabnzbd /bin/bash
root@4d549f2f6a9c:/# find / | grep getipaddress.py
/app/sabnzbd/tests/test_getipaddress.py
/app/sabnzbd/sabnzbd/getipaddress.py
root@4d549f2f6a9c:/# cd /app/sabnzbd/
root@4d549f2f6a9c:/app/sabnzbd# python3
Python 3.9.7 (default, Nov 24 2021, 21:15:59) 
[GCC 10.3.1 20211027] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sabnzbd.getipaddress import *
>>> addresslookup4('self-test.sabnzbd.org')
[(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, 'self-test.sabnzbd.org', ('104.26.2.117', 80)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_DGRAM: 2>, 17, 'self-test.sabnzbd.org', ('104.26.2.117', 80)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, 'self-test.sabnzbd.org', ('172.67.69.176', 80)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_DGRAM: 2>, 17, 'self-test.sabnzbd.org', ('172.67.69.176', 80)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, 'self-test.sabnzbd.org', ('104.26.3.117', 80)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_DGRAM: 2>, 17, 'self-test.sabnzbd.org', ('104.26.3.117', 80))]
>>> publicipv4()
'12.12.162.65'
>>> 

```

**Comment by RafaelSchridi** [comment]
> @RafaelSchridi what is your setup? Like: is your SAB in docker, or on a plain OS?

I'm using docker on a fresh arch install
```
services:
  sabnzbd:
    container_name: sabnzbd
    restart: unless-stopped
    volumes:
      - ${DOCKER_SETTINGS_DIR}/sabnzbd:/config
      - ${DOCKER_FAST_CACHE_DIR}:/downloads
    environment:
      - PUID=${PUID}
      - PGID=${PGID}
      - TZ=${TZ}
      - HAS_IPV6=false
      - UMASK=007
    ports:
      - 8080:8080
    image: linuxserver/sabnzbd
```


> both of you:
> 
> what if you visit https://self-test.sabnzbd.org/ with your web browser? what is the result of `host self-test.sabnzbd.org` or `nslookup self-test.sabnzbd.org`

If i visit on my browser it shows my ip adress
And inside the container 
```
root@3e53e7476c72:/#  nslookup self-test.sabnzbd.org
Server:         127.0.0.11
Address:        127.0.0.11:53

Non-authoritative answer:
Name:   self-test.sabnzbd.org
Address: 2606:4700:20::ac43:45b0
Name:   self-test.sabnzbd.org
Address: 2606:4700:20::681a:275
Name:   self-test.sabnzbd.org
Address: 2606:4700:20::681a:375

Non-authoritative answer:
Name:   self-test.sabnzbd.org
Address: 172.67.69.176
Name:   self-test.sabnzbd.org
Address: 104.26.2.117
Name:   self-test.sabnzbd.org
Address: 104.26.3.117
```

And for the last thing, i executed it in my already running container
```
root@3e53e7476c72:/# find / | grep getipaddress.py
find: /proc/315/task/315/fdinfo: Permission denied
find: /proc/315/task/338/fdinfo: Permission denied
find: /proc/315/task/339/fdinfo: Permission denied
find: /proc/315/task/340/fdinfo: Permission denied
find: /proc/315/task/341/fdinfo: Permission denied
find: /proc/315/task/342/fdinfo: Permission denied
find: /proc/315/task/343/fdinfo: Permission denied
find: /proc/315/task/344/fdinfo: Permission denied
find: /proc/315/task/345/fdinfo: Permission denied
find: /proc/315/task/346/fdinfo: Permission denied
find: /proc/315/task/347/fdinfo: Permission denied
find: /proc/315/task/348/fdinfo: Permission denied
find: /proc/315/task/349/fdinfo: Permission denied
find: /proc/315/task/350/fdinfo: Permission denied
find: /proc/315/task/351/fdinfo: Permission denied
find: /proc/315/task/352/fdinfo: Permission denied
find: /proc/315/task/353/fdinfo: Permission denied
find: /proc/315/task/355/fdinfo: Permission denied
find: /proc/315/task/356/fdinfo: Permission denied
find: /proc/315/task/357/fdinfo: Permission denied
find: /proc/315/task/358/fdinfo: Permission denied
find: /proc/315/task/363/fdinfo: Permission denied
find: /proc/315/fdinfo: Permission denied
/app/sabnzbd/sabnzbd/getipaddress.py
/app/sabnzbd/tests/test_getipaddress.py
root@3e53e7476c72:/# cd /app/sabnzbd/
root@3e53e7476c72:/app/sabnzbd# python3
Python 3.9.7 (default, Nov 24 2021, 21:15:59) 
[GCC 10.3.1 20211027] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sabnzbd.getipaddress import *
>>> addresslookup4('self-test.sabnzbd.org')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/app/sabnzbd/sabnzbd/getipaddress.py", line 48, in func_wrapper
    return sabnzbd.THREAD_POOL.submit(item, *args, **kwargs).result(max_timeout)
  File "/usr/lib/python3.9/concurrent/futures/_base.py", line 447, in result
    raise TimeoutError()
concurrent.futures._base.TimeoutError
```

I've got a feeling this might be a ipv6 thing, i've had problems with ipv6 and sabnzbd before: https://github.com/linuxserver/docker-sabnzbd/issues/82

**Comment by RafaelSchridi** [comment]
OH using a fresh container works
 ```
[rafael@sol /home/rafael]$ sudo docker exec -it sabnzbdtest /bin/bash
root@f66d58582038:/# find / | grep getipaddress.py
find: /proc/302/task/302/fdinfo: Permission denied
find: /proc/302/task/323/fdinfo: Permission denied
find: /proc/302/task/324/fdinfo: Permission denied
find: /proc/302/task/325/fdinfo: Permission denied
find: /proc/302/task/326/fdinfo: Permission denied
find: /proc/302/task/327/fdinfo: Permission denied
find: /proc/302/task/328/fdinfo: Permission denied
find: /proc/302/task/329/fdinfo: Permission denied
find: /proc/302/task/330/fdinfo: Permission denied
find: /proc/302/task/331/fdinfo: Permission denied
find: /proc/302/task/332/fdinfo: Permission denied
find: /proc/302/task/333/fdinfo: Permission denied
find: /proc/302/task/334/fdinfo: Permission denied
find: /proc/302/task/335/fdinfo: Permission denied
find: /proc/302/task/337/fdinfo: Permission denied
find: /proc/302/task/338/fdinfo: Permission denied
find: /proc/302/task/339/fdinfo: Permission denied
find: /proc/302/task/340/fdinfo: Permission denied
find: /proc/302/task/341/fdinfo: Permission denied
find: /proc/302/task/342/fdinfo: Permission denied
find: /proc/302/task/343/fdinfo: Permission denied
find: /proc/302/task/344/fdinfo: Permission denied
find: /proc/302/fdinfo: Permission denied
/app/sabnzbd/sabnzbd/getipaddress.py
/app/sabnzbd/tests/test_getipaddress.py
root@f66d58582038:/# cd /app/sabnzbd/
root@f66d58582038:/app/sabnzbd# python3
Python 3.9.7 (default, Nov 24 2021, 21:15:59) 
[GCC 10.3.1 20211027] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sabnzbd.getipaddress import *
>>> addresslookup4('self-test.sabnzbd.org')
[(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, 'self-test.sabnzbd.org', ('104.26.2.117', 80)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_DGRAM: 2>, 17, 'self-test.sabnzbd.org', ('104.26.2.117', 80)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, 'self-test.sabnzbd.org', ('104.26.3.117', 80)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_DGRAM: 2>, 17, 'self-test.sabnzbd.org', ('104.26.3.117', 80)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, 'self-test.sabnzbd.org', ('172.67.69.176', 80)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_DGRAM: 2>, 17, 'self-test.sabnzbd.org', ('172.67.69.176', 80))]
```

**Comment by herkalurk** [comment]
I can execute the commands successfully on a fresh container, but on my existing install it doesn't work like @RafaelSchridi 

```
[root@SERVER ~]# docker exec -it sabnzbd /bin/bash
root@76d86558d606:/# cd /app/sabnzbd/
root@76d86558d606:/app/sabnzbd# python3
Python 3.9.7 (default, Nov 24 2021, 21:15:59)
[GCC 10.3.1 20211027] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sabnzbd.getipaddress import *
>>> addresslookup4('self-test.sabnzbd.org')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/app/sabnzbd/sabnzbd/getipaddress.py", line 48, in func_wrapper
    return sabnzbd.THREAD_POOL.submit(item, *args, **kwargs).result(max_timeout)
  File "/usr/lib/python3.9/concurrent/futures/_base.py", line 447, in result
    raise TimeoutError()
concurrent.futures._base.TimeoutError
```

**Comment by herkalurk** [comment]
> I've got a feeling this might be a ipv6 thing, i've had problems with ipv6 and sabnzbd before: [linuxserver/docker-sabnzbd#82](https://github.com/linuxserver/docker-sabnzbd/issues/82)

I've wanted to turn off IPv6 in docker for some time, haven't found a good way to do so.

**Comment by herkalurk** [comment]
> @herkalurk are you using PiHole too?

I am using a different but similar docker container called adguard.  The logs show it processed the dns queries successfully though.  Log screenshot below showing the gateway from the docker container network.

![image](https://user-images.githubusercontent.com/2532699/172724858-cb3fd3ef-05a1-4998-b484-ec560d1b8f82.png)

**Comment by herkalurk** [comment]
> EDIT: Ah, that message is a side effect of docker and port mapping: SAB thinks it's on 8080 and checks that, but docker's port mapping like for example `-p 8085:8080` has it mapped on 8085.

The whole config is to run on port 8085, not just a port mapping.
```
version: "2.1"
services:
  sabnzbd:
    image: linuxserver/sabnzbd
    container_name: sabnzbd
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=America/Chicago
    volumes:
      - /home/docker/sabnzbd:/config
      - /media/usenet:/media/usenet
      - /media/movies:/media/movies
      - /media/tv:/media/tv
    ports:
      - 8085:8085
    restart: unless-stopped
networks:
  default:
    external:
      name: sickly
```

**Comment by sanderjo** [comment]
So @herkalurk @RafaelSchridi have a few things in common:

- in Docker image linuxserver/sabnzbd the problem occurs (but not in a fresh image)?
- some special DNS setup (pihole, adguard)

So are you able to reduce and deduce what the cause is? For example: use direct dns (like 1.1.1.1) instead of pihole/adguard?

Also: what happens if you use api.ipify.org (https://api.ipify.org) as self-test host in SABnzbd?

**Comment by RafaelSchridi** [comment]
api.ipify.org does the same thing, but I think I figured it out, it IS a DNS issue, forcing 1.1.1.1 on the container fixes it.
```yaml
services:
  sabnzbd:
    container_name: sabnzbd
    restart: unless-stopped
    volumes:
      - ${DOCKER_SETTINGS_DIR}/sabnzbd:/config
      - ${DOCKER_FAST_CACHE_DIR}:/downloads
    environment:
      - PUID=${PUID}
      - PGID=${PGID}
      - TZ=${TZ}
      - HAS_IPV6=false
      - UMASK=007
    ports:
      - 8080:8080
    dns: 
      - 1.1.1.1
      - 8.8.8.8
    image: linuxserver/sabnzbd
```
![image](https://user-images.githubusercontent.com/11460682/172837780-ba9a9f05-b9ea-405b-b055-0874aae17d67.png)

**Comment by RafaelSchridi** [comment]
I think this might be some docker fuckery, because replace 1.1.1.1 with my pihole IP also works.

**Comment by sanderjo** [comment]
> but I think I figured it out, it IS a DNS issue, forcing 1.1.1.1 on the container fixes it.

Great finding. 

So ... does pihole / adguard give back a malformed DNS response (as it seems not understood within python/sabnzbd)? Or no response within the 3 second timeout? 

This is the lookup within SABnzbd that fails ... pure python code:

```
@timeout(3.0)
def addresslookup4(myhost):
    return socket.getaddrinfo(myhost, 80, socket.AF_INET)
```

That fails (Timeout) against pihole / adguard. Can you reproduce that? Most simple python code:

```
import socket
print(socket.getaddrinfo("self-test.sabnzbd.org", 80, socket.AF_INET))
```

On my Ubuntu, no docker, no pihole:

```
sander@brixit:~$ python3
Python 3.8.10 (default, Mar 15 2022, 12:22:08)
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import socket
>>> print(socket.getaddrinfo("self-test.sabnzbd.org", 80, socket.AF_INET))
[(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('172.67.69.176', 80)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_DGRAM: 2>, 17, '', ('172.67.69.176', 80)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_RAW: 3>, 0, '', ('172.67.69.176', 80)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('104.26.3.117', 80)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_DGRAM: 2>, 17, '', ('104.26.3.117', 80)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_RAW: 3>, 0, '', ('104.26.3.117', 80)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('104.26.2.117', 80)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_DGRAM: 2>, 17, '', ('104.26.2.117', 80)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_RAW: 3>, 0, '', ('104.26.2.117', 80))]
>>>
```

**Comment by herkalurk** [comment]
> @herkalurk so what do you think the root cause is? Please consider [#2200 (comment)](https://github.com/sabnzbd/sabnzbd/issues/2200#issuecomment-1151012397) ... with plain DNS, it just works.

Based on everything shown the root cause appears to be legacy settings in the `sabnzbd.ini` file.  Both myself and @RafaelSchridi have shown on an existing config we have issues with timeout, however on a brand new container with no config and **NO** changes to container specific dns settings everything works as expected.  In my testing I didn't change anything about my docker-compose file or container, I simply removed everything from the `/config` directory and restarted the container to have it function as expected.  Something in the config is causing this.  

I've been running sabnzbd since at least 2014 (not as a docker container until about a year ago), my config file has been changing as your team has been changing/updating the app all this time.  My config has 528 lines, what is the current base config like?  I can pull out the private info in my config and share to dig further into this if we need to.

**Comment by sanderjo** [comment]
> however on a brand new container

Wait: do you mean a brand new **SABnzbd** container, with pihole/adguard?

> My config has 528 lines

I wanted to say "big", but I see my sabnzbd.ini is 810 lines (a few different newsservers and indexers)

You could do a `diff` of your real sabnzbd.ini and the fresh sabnzbd.ini, and try to find the difference that's causing it.

**Comment by RafaelSchridi** [comment]
I think i fixed it completely, I have no more DNS providers defined in my sab docker-compose, but i'm still not sure why this is working:

I put my Pihole container in network:host mode. I think there problem is somewhere between 2 docker bridge networks, sab was able to reach pihole but i don't think it the reponse ever arrived. And for some reason this was only a problem in 1 of my 35 containers.

heres my pihole config now:
```yaml
  pihole:
    container_name: pihole
    hostname: pihole01.local
    image: pihole/pihole:latest
    environment:
      TZ: ${TZ}
      DHCP_ACTIVE: false
      WEB_PORT: 8082
    volumes:
      - ${DOCKER_SETTINGS_DIR}/pihole/etc-pihole/:/etc/pihole/
      - ${DOCKER_SETTINGS_DIR}/pihole/etc-dnsmasq.d/:/etc/dnsmasq.d/
    cap_add:
      - CAP_SYS_NICE
      - CAP_NET_BIND_SERVICE
    restart: unless-stopped
    network_mode: host
```

**Comment by herkalurk** [comment]
> Wait: do you mean a brand new **SABnzbd** container, with pihole/adguard?

Yes, if we created a duplicate test sabnzbd container with no config and not adjusting any dns like `docker run -d --name sabnzbd_test -p 8080:8080 linuxserver/sabnzbd` those worked correctly.

**Comment by Safihre** [comment]
Closing, as crash is fixed by #2202
